test/controllers/contactPreference/ChangeEmailControllerSpec.scala

```diff
@@ -0,0 +1,336 @@
+            mockVatSubscriptionService,
+            mockErrorHandler,
+            mockAuditingService,
+            view
+          )
+        }
+
+        lazy val result = inflightTarget.show(request)
+
+        "return 200" in {
+          status(result) shouldBe Status.OK
+        }
+
+        "return HTML" in {
+          contentType(result) shouldBe Some("text/html")
+          charset(result) shouldBe Some("utf-8")
+        }
+      }
+    }
+    " letterToConfirmedEmail feature switch is off" should {
+      lazy val result = {
+        mockConfig.features.letterToConfirmedEmailEnabled(false)
+        target().show(request
+          .withSession(common.SessionKeys.validationEmailKey -> testValidationEmail))
+      }
+      "return NOT FOUND" in {
+        status(result) shouldBe Status.NOT_FOUND
+      }
+
+    }
+  }
+
+  "Calling the submit action" when {
+
+    "the letterToConfirmedEmail feature switch is turned on" when {
+
+      "a user is enrolled with a valid enrolment" when {
+
+        "there is an email in session" when {
+
+          "the form is successfully submitted" should {
+
+            lazy val result = {
+              mockConfig.features.letterToConfirmedEmailEnabled(true)
+              target().submit(request
+                .withFormUrlEncodedBody("email" -> testValidEmail)
+                .withSession(common.SessionKeys.validationEmailKey -> testValidationEmail))
+            }
+            "redirect to the confirm email view" in {
+              status(result) shouldBe Status.SEE_OTHER
+              redirectLocation(result) shouldBe Some("")
+            }
+
+            "add the new email to the session" in {
+              session(result).get(SessionKeys.prepopulationEmailKey) shouldBe Some(testValidEmail)
+            }
+          }
+
+          "the form is unsuccessfully submitted" should {
+
+            lazy val result = {
+              mockConfig.features.letterToConfirmedEmailEnabled(true)
+              target().submit(request
+                .withFormUrlEncodedBody("email" -> testInvalidEmail)
+                .withSession(common.SessionKeys.validationEmailKey -> testValidationEmail))
+            }
+
+            "reload the page with errors" in {
+              status(result) shouldBe Status.BAD_REQUEST
+            }
+
+            "return HTML" in {
+              contentType(result) shouldBe Some("text/html")
+              charset(result) shouldBe Some("utf-8")
+            }
+          }
+        }
+
+        "there is no email in session" when {
+
+          lazy val result = target().submit(request)
+
+          "render the error view" in {
+            status(result) shouldBe Status.INTERNAL_SERVER_ERROR
+          }
+
+          "return HTML" in {
+            contentType(result) shouldBe Some("text/html")
+            charset(result) shouldBe Some("utf-8")
+          }
+        }
+      }
+
+      "a user is does not have a valid enrolment" should {
```

Comment:
Grammar - "a user is does"

Reply:
I know this is tiny but it wasn't addressed the first time around, and while you're here again, would you mind?